aws/resource_aws_sagemaker_lifecycle_configuration.go

```diff
@@ -0,0 +1,168 @@
+	conn := meta.(*AWSClient).sagemakerconn
+
+	request := &sagemaker.DescribeNotebookInstanceLifecycleConfigInput{
+		NotebookInstanceLifecycleConfigName: aws.String(d.Id()),
+	}
+
+	lifecycleConfig, err := conn.DescribeNotebookInstanceLifecycleConfig(request)
+	if err != nil {
+		if isAWSErr(err, "ValidationException", "") {
+			log.Printf("[INFO] unable to find the SageMaker notebook instance lifecycle configuration (%s); therefore it is removed from the state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error reading SageMaker notebook instance lifecycle configuration %s: %s", d.Id(), err)
+	}
+
+	if err := d.Set("name", lifecycleConfig.NotebookInstanceLifecycleConfigName); err != nil {
+		return fmt.Errorf("error setting name for SageMaker notebook instance lifecycle configuration (%s): %s", d.Id(), err)
+	}
+
+	if len(lifecycleConfig.OnCreate) > 0 {
+		if err := d.Set("on_create", lifecycleConfig.OnCreate[0].Content); err != nil {
+			return fmt.Errorf("error setting on_create for SageMaker notebook instance lifecycle configuration (%s): %s", d.Id(), err)
+		}
+	}
+
+	if len(lifecycleConfig.OnStart) > 0 {
+		if err := d.Set("on_start", lifecycleConfig.OnStart[0].Content); err != nil {
+			return fmt.Errorf("error setting on_start for SageMaker notebook instance lifecycle configuration (%s): %s", d.Id(), err)
+		}
+	}
+
+	if err := d.Set("arn", lifecycleConfig.NotebookInstanceLifecycleConfigArn); err != nil {
+		return fmt.Errorf("error setting arn for SageMaker notebook instance lifecycle configuration (%s): %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func resourceAwsSagemakerLifeCycleConfigurationUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	updateOpts := &sagemaker.UpdateNotebookInstanceLifecycleConfigInput{
+		NotebookInstanceLifecycleConfigName: aws.String(d.Get("name").(string)),
+	}
+
+	onCreateHook := &sagemaker.NotebookInstanceLifecycleHook{Content: aws.String(d.Get("on_create").(string))}
+	updateOpts.SetOnCreate([]*sagemaker.NotebookInstanceLifecycleHook{onCreateHook})
+
+	onStartHook := &sagemaker.NotebookInstanceLifecycleHook{Content: aws.String(d.Get("on_start").(string))}
+	updateOpts.SetOnStart([]*sagemaker.NotebookInstanceLifecycleHook{onStartHook})
+
+	_, err := conn.UpdateNotebookInstanceLifecycleConfig(updateOpts)
+	if err != nil {
+		return fmt.Errorf("error updating SageMaker notebook instance lifecycle configuration: %s", err)
+	}
+	return resourceAwsSagemakerLifeCycleConfigurationRead(d, meta)
+}
+
+func resourceAwsSagemakerLifeCycleConfigurationDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	deleteOpts := &sagemaker.DeleteNotebookInstanceLifecycleConfigInput{
+		NotebookInstanceLifecycleConfigName: aws.String(d.Id()),
+	}
+	log.Printf("[INFO] Deleting SageMaker notebook instance lifecycle configuration: %s", d.Id())
+
+	return resource.Retry(5*time.Minute, func() *resource.RetryError {
```

Comment:
Since this `resource.Retry()` contains no `return resource.RetryableError()` conditions, the `resource.Retry()` logic should be removed.